File: maggioliEbookApp/iosMaggioliEbookApp/iosMaggioliEbookApp/Common/Extensions/Bookmark.swift
//
//  Bookmark.swift
//  iosMaggioliEbookApp
//
//  Created by Filippo Paganelli on 17/10/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import shared
import R2Shared

extension Bookmark: Identifiable {
    
    var positionText: String? {
        if let position = self.toLocator().locations.position {
            return String(format: NSLocalizedString("reader_outline_position_label", comment: "Outline bookmark label when the position is available"), position)
        } else if let progression = self.toLocator().locations.progression {
            return String(format: NSLocalizedString("reader_outline_progression_label", comment: "Outline bookmark label when the progression is available"), progression * 100)
        } else {
            return nil
        }
    }
    
    ///
    func toLocator() -> Locator {
        return Locator(href: self.resourceHref,
                       type: self.resourceType,
                       title: self.resourceTitle,
                       locations: Locator.Locations(jsonString: self.location),
                       text: Locator.Text(jsonString: self.locatorText)
        )
    }
}

File: maggioliEbookApp/androidMaggioliEbookApp/src/main/java/it/filo/maggioliebook/android/utils/extensions/Bookmark.kt
package it.filo.maggioliebook.android.utils.extensions

import it.filo.maggioliebook.domain.core.book.Bookmark
import org.json.JSONObject
import org.readium.r2.shared.publication.Locator

fun Bookmark.toLocator(): Locator = Locator(
    href = resourceHref,
    type = resourceType,
    title = resourceTitle,
    locations = Locator.Locations.fromJSON(JSONObject(location)),
    text = Locator.Text.fromJSON(JSONObject(locatorText))
)